### ISSUE TITLE
ExifTool.Tag does not add a type hint

### ISSUE THREAD

**Issue by ghost** [rkalla/imgscalr#37]
In the current implementation of ExifTool.Tag, no type information is stored along side the Tag or it's name; it might be handy for users if a type is stored along with the name to make parsing for them easier.

If -n is not passed to ExifTool, then the long-form/friendly names for values are used instead which will no longer make the type valid.

To account for that flexibility in a safe way, the type provided should be used more as a hint and not a defactor do-or-die type conversion. This will allow people working with custom tags or custom types (maybe an older or newer version of ExifTool down the road?) to account for changes and parse things themselves without the tool exploding in a heap of fire.

Basically, plan and design for flexibility so the world doesn't melt if the ExifTool behavior changes in the future.
